File: main/src/main/java/com/sedmelluq/lavaplayer/core/source/youtube/YoutubeSearchResultLoader.java
package com.sedmelluq.lavaplayer.core.source.youtube;

import com.sedmelluq.lavaplayer.core.http.ExtendedHttpConfigurable;
import com.sedmelluq.lavaplayer.core.info.AudioInfoEntity;
import com.sedmelluq.lavaplayer.core.info.track.AudioTrackInfoTemplate;

public interface YoutubeSearchResultLoader {
  AudioInfoEntity loadSearchResult(String query, AudioTrackInfoTemplate template);

  ExtendedHttpConfigurable getHttpConfiguration();
}

File: main/src/main/java/com/sedmelluq/discord/lavaplayer/source/youtube/YoutubeSearchResultLoader.kt
package com.sedmelluq.discord.lavaplayer.source.youtube

import com.sedmelluq.discord.lavaplayer.tools.http.ExtendedHttpConfigurable
import com.sedmelluq.discord.lavaplayer.track.AudioItem
import com.sedmelluq.discord.lavaplayer.track.AudioTrackFactory

interface YoutubeSearchResultLoader {
    val httpConfiguration: ExtendedHttpConfigurable

    fun loadSearchResult(query: String, trackFactory: AudioTrackFactory): AudioItem
}
